android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -886,6 +893,41 @@ public static boolean setKeyboard(String packageID, String keyboardID, String la
     return (result1 || result2);
   }
 
+  /**
+   * Prepare keyboard switch for inapp keyboard and systemkeyboard
+   * @param packageID the package id
+   * @param keyboardID the keyboard id
+   * @param languageID the language id
+   * @param keyboardName keyboard name
+   * @return the success result
+   */
+  public static boolean prepareKeyboardSwitch(String packageID, String keyboardID, String languageID, String keyboardName) {
+
+    //reset banner and layout params
+    currentBanner = KMManager.KM_BANNER_STATE_BLANK;
```

Comment:
The keyboard loading is a lot better now!

Ideally, if we're switching from a keyboard with dictionary (e.g. sil_euro_latin) to another one with dictionary (e.g. fv_sencoten), we wouldn't have this "blank" banner flicker in between. It can wait for another day though.